pages/reference/OS/secure-boot-and-full-disk-encryption/overview.md

```diff
@@ -0,0 +1,121 @@
+
+
+__Note:__ This imposes a limitation on the specifications for device types to support secure boot - they must have enough memory to hold the raw image that is being programmed into disk. BalenaOS reduces this restriction by using compressed memory to hold the raw image, but still, memory constrained devices will not be able to support secure boot with the current design.
+
+
+### Partition Layout and Disk Encryption
+Balena uses the `dm-crypt` Linux kernel subsystem and `device-mapper` framework to encrypt block devices. Both direct `dm-crypt` and `LUKS` (Linux Unified Key Setup) are supported.
+
+A secure boot installation also alters the default balenaOS partition table as the boot partition is split into two:
+- An unencrypted boot partition which contains the signed balena bootloader binary and other essential artifacts needed to run it.
+- An encrypted boot partition that mounts in the traditional `/mnt/boot` path and contains all other artifacts, including confidential files like `config.json` which contains API keys.
+
+
+## Boot Chain of Trust
+
+The notion of a chain of trust arises because each stage in the boot process validates the integrity of the subsequent stage before executing it. If at any point a link in this chain is corrupted, unsigned, or otherwise fails validation, the boot process halts or reverts to a known-good state. This sequential validation ensures that if the first link is trusted, every subsequent link remains trustworthy, preventing unauthorized or tampered code from ever running.
+
+The root of trust is always the device’s `bootROM`. `bootROM` refers to the very first piece of firmware that runs when a device powers on. It resides in read-only memory (ROM) that is physically embedded in the processor or system-on-chip (SoC). Because the code stored in bootROM is difficult or impossible to modify (without physically altering the chip), it serves as the hardware-enforced starting point or “root of trust” for the boot process. 
+
+Key responsibilities of a bootROM include:
+
+- **Initializing critical hardware**: Setting up CPU registers, clocks, and memory controllers in order to begin loading higher-level software.
+- **Locating and verifying the next boot stage**: Often checks for cryptographic signatures or checksums on the bootloader (or other initial executables) to ensure they have not been tampered with.
+- **Passing control to a validated bootloader**: Once signature verification succeeds, execution is handed off to the bootloader, continuing the chain of trust.
+
+A typical boot chain of trust for balenaOS involves:
+- The `bootROM` validates the balena bootloader and device trees and executes the balena bootloader
+- The balena bootloader mounts the encrypted storage, validates the appropriate Linux kernel plus initramfs using in-kernel X.509 certificates
```

Comment:
in the last bullet below we put X.509 in code quotes but here we don't, we should be consistent.